### PR TITLE
hardware-and-software-requirements: explicitly exclude Ubuntu 21.10

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Explictly exclude Ubuntu 21.10 from the list of supported operating systems as current TiFlash versions can't work with new glibc versions.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

**Comment by dveeden** [comment]
Note that the latest Ubuntu LTS release is 20.04 and the latest non-LTS release is 21.10. See https://releases.ubuntu.com/
Maybe we should rephrase this to be less confusing.
Note that Ubuntu 22.04 is going to be an LTS release (source: https://ubuntu.com/about/release-cycle ), but would probably also need to be excluded from the list of supported versions.

**Comment by morgo** [comment]
@dveeden Is there a known TiFlash bug we can point to so users will know when this is resolved? (or documenters; since we rarely track supported platforms diligently).

I am an Ubuntu 21.10 user, I was wondering why TiFlash was broken.

**Comment by dveeden** [comment]
> @dveeden Is there a known _public_ TiFlash bug we can point to so users will know when this is resolved?

I'm not aware of any public bugs that cover this.

**Comment by morgo** [comment]
> I'm not aware of any public bugs that cover this.

Can we link to this PR/issue since it has a little more context.

**Review by SchrodingerZhu — APPROVED** [review]
LGTM. According to previous tests by @dveeden, it is true that TiFlash has compatibility issues with newer glibcs (for all released stable versions; but not those nightly ones).

**Comment by dveeden** [comment]
With v6.x releases of TiFlash it is no longer needed to exclude Ubuntu 21.10
